code/edgeidentity/src/main/java/com/adobe/marketing/mobile/edge/identity/EventUtils.java

```diff
@@ -0,0 +1,115 @@
+	}
+
+	/**
+	 * Checks if the provided {@code event} is of type {@link IdentityConstants.EventType#EDGE_IDENTITY} and source {@link IdentityConstants.EventSource#UPDATE_IDENTITY}
+	 *
+	 * @param event the event to verify
+	 * @return true if both type and source match
+	 */
+	static boolean isUpdateIdentityEvent(final Event event) {
+		return event != null && IdentityConstants.EventType.EDGE_IDENTITY.equalsIgnoreCase(event.getType())
+			   && IdentityConstants.EventSource.UPDATE_IDENTITY.equalsIgnoreCase(event.getSource());
+	}
+
+	/**
+	 * Checks if the provided {@code event} is of type {@link IdentityConstants.EventType#EDGE_IDENTITY} and source {@link IdentityConstants.EventSource#REQUEST_IDENTITY}
+	 *
+	 * @param event the event to verify
+	 * @return true if both type and source match
+	 */
+	static boolean isRequestIdentityEvent(final Event event) {
+		return event != null && IdentityConstants.EventType.EDGE_IDENTITY.equalsIgnoreCase(event.getType())
+			   && IdentityConstants.EventSource.REQUEST_IDENTITY.equalsIgnoreCase(event.getSource());
+	}
+
+	/**
+	 * Checks if the provided {@code event} is of type {@link IdentityConstants.EventType#GENERIC_IDENTITY} and source {@link IdentityConstants.EventSource#REQUEST_RESET}
+	 *
+	 * @param event the event to verify
+	 * @return true if both type and source match
+	 */
+	static boolean isRequestResetEvent(final Event event) {
+		return event != null && IdentityConstants.EventType.GENERIC_IDENTITY.equalsIgnoreCase(event.getType())
+			   && IdentityConstants.EventSource.REQUEST_RESET.equalsIgnoreCase(event.getSource());
+	}
+
+	/**
+	 * Checks if the provided {@code event} is a shared state update event for {@code stateOwnerName}
+	 *
+	 * @param stateOwnerName the shared state owner name; should not be null
+	 * @param event current event to check; should not be null
+	 * @return {@code boolean} indicating if it is the shared state update for the provided {@code stateOwnerName}
+	 */
+	static boolean isSharedStateUpdateFor(final String stateOwnerName, final Event event) {
+		if (Utils.isNullOrEmpty(stateOwnerName) || event == null) {
+			return false;
+		}
+
+		String stateOwner;
+
+		try {
+			stateOwner = (String) event.getEventData().get(IdentityConstants.SharedState.STATE_OWNER);
+		} catch (ClassCastException e) {
+			return false;
+		}
+
+		return stateOwnerName.equals(stateOwner);
+	}
+
+	/**
+	 * Extracts the ECID from the Identity Direct shared state and returns it as an {@link ECID} object
+	 *
+	 * @param identityDirectSharedState the Identity Direct shared state data
+	 * @return the ECID or null if not found or unable to parse the payload
+	 */
+	static ECID getECID(final Map<String, Object> identityDirectSharedState) {
+		ECID legacyEcid = null;
+
+		try {
+			final String legacyEcidString = (String) identityDirectSharedState.get(
+												IdentityConstants.SharedState.IdentityDirect.ECID);
+			legacyEcid = legacyEcidString == null ? null : new ECID(legacyEcidString);
```

Comment:
I would suggest not generating a new ECID here, just return null so the caller knows the ECID didn't exist in the shared state. In the code where this API is called, you check if `legacyEcid == null` and if not you set the Edge Identity ECID to this returned ECID and log something to the effect `ECID migrated from direct Identity`. However, if the shared state contains no ECID and you generate on here, then the log message is inaccurate as the ECID wasn't migrated.

Reply:
this would return null if the legacyEcidString is null or unable to cast

Reply:
Yes, I see that now. Sorry, I misread the code. This looks good. 👍